juntagrico/views_create_subscription.py

```diff
@@ -89,63 +91,80 @@ def cs_add_member(request, main_member):
     return render(request, 'createsubscription/add_member_cs.html', renderdict)
 
 
-@requires_main_member
-def cs_select_shares(request, main_member):
-    co_members = request.session.get('create_co_members', [])
+class CSSelectSharesView(TemplateView):
```

Comment:
i need some more time to decide how i feel about this, but looks promising

Reply:
You will love it. this reduced cognitive complexity from 49 to below 15.
https://codeclimate.com/github/juntagrico/juntagrico/pull/85

Reply:
that part in fact i do love, what i am sceptical is that we would introduce another pattern. i guess we could achieve the cofnitive complexity reduction using the helper methods without the class.
on the other hand it is a powerfull pattern and maybe we should start to use it everywhere, which would mean a huge refactoring of the code. 
also this is less straitfoard than a simple method so i do no know which pattern is better for new contributors